exporter/clickhouseexporter/config.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clickhouseexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/clickhouseexporter"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.uber.org/multierr"
+)
+
+// Config defines configuration for Elastic exporter.
+type Config struct {
+	config.ExporterSettings        `mapstructure:",squash"`
+	exporterhelper.TimeoutSettings `mapstructure:",squash"`
+	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
+	// QueueSettings is a subset of exporterhelper.QueueSettings,
+	// because only QueueSize is user-settable.
+	QueueSettings QueueSettings `mapstructure:"sending_queue"`
+
+	// DSN is the ClickHouse server Data Source Name.
+	// For tcp protocol reference: [ClickHouse/clickhouse-go#dsn](https://github.com/ClickHouse/clickhouse-go#dsn).
+	// For http protocol reference: [mailru/go-clickhouse/#dsn](https://github.com/mailru/go-clickhouse/#dsn).
+	DSN string `mapstructure:"dsn"`
+	// TTLDays is The data time-to-live in days, 0 means no ttl.
+	TTLDays uint `mapstructure:"ttl_days"`
+}
+
+// QueueSettings is a subset of exporterhelper.QueueSettings.
+type QueueSettings struct {
+	// QueueSize set the length of the sending queue
+	QueueSize int `mapstructure:"queue_size"`
+}
+
+var (
+	errConfigNoDSN = errors.New("dsn must be specified")
+)
+
+// Validate validates the clickhouse server configuration.
+func (cfg *Config) Validate() (err error) {
+	if cfg.DSN == "" {
+		err = multierr.Append(err, errConfigNoDSN)
+	}
+	return err
+}
+
+func (cfg *Config) enforcedQueueSettings() exporterhelper.QueueSettings {
+	return exporterhelper.QueueSettings{
+		Enabled:      true,
+		NumConsumers: 1,
+		QueueSize:    cfg.QueueSettings.QueueSize,
+	}
+}
```

Comment:
This may be the reason for failing linter

Reply:
Thank you, Fixed.